src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/HdcedCalculator.kt

```diff
@@ -135,6 +146,46 @@ class HdcedCalculator(
     }
   }
 
+  // This is a copy of calculateHdcedUnderMidpoint - the only difference being it saves against the new HDC-365 variant and uses it's config
+  // Duplicated the method rather than modified it with switches as the other method can just be deleted after the HDC365 commencement date
+  private fun calculateHdcedUnderMidpointHDC365(
+    sentenceCalculation: SentenceCalculation,
+    sentence: CalculableSentence,
+    params: HdcedParams,
+  ) {
+    val halfTheCustodialPeriodButAtLeastTheMinimumHDCEDPeriod = max(
+      hdcedConfiguration.custodialPeriodBelowMidpointMinimumDeductionDays,
+      ceil(params.custodialPeriod.div(HALF)).toLong(),
+    )
+
+    sentenceCalculation.numberOfDaysToHomeDetentionCurfewEligibilityDateHDC365 =
+      halfTheCustodialPeriodButAtLeastTheMinimumHDCEDPeriod.plus(params.adjustedDays)
+    sentenceCalculation.homeDetentionCurfewEligibilityDateHDC365 =
+      params.dateHdcAppliesFrom.plusDays(sentenceCalculation.numberOfDaysToHomeDetentionCurfewEligibilityDateHDC365)
+
+    if (isCalculatedHdc365LessThanTheMinimumHDCPeriod(sentence, sentenceCalculation, params)) {
+      calculateHdcedMinimumCustodialPeriodHDC365(
+        sentence,
+        sentenceCalculation,
+        CalculationRule.HDCED_GE_MIN_PERIOD_LT_MIDPOINT,
+        params,
+      )
+    } else {
+      sentenceCalculation.breakdownByReleaseDateType[ReleaseDateType.HDCED365] =
```

Comment:
Would prefer to avoid HDCED365 in the release date types if possible as release date types are needed for historic view journeys, and therefore cannot be removed see HDC4+

Reply:
I've reworked this to use a different (new) enum - let me know what you think